### PR TITLE
Highlights and load

### PR DESCRIPTION
**Pull request by ahgittin**
uses https://github.com/apache/brooklyn-server/pull/818 to add highlights for software-specific entities.  also adds new perf test.

### CODE REVIEW

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java [hunk 1]
```diff
@@ -56,12 +60,17 @@ public void start(Collection<? extends Location> locations) {
     
     public static class MemberTrackingPolicy extends AbstractMembershipTrackingPolicy {
         @Override protected void onEntityEvent(EventType type, Entity member) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: Not needed as this policy extends `AbstractMembershipTrackingPolicy ` and `onEntityEvent` [already call `defaultHighlightAction`](https://github.com/apache/brooklyn-server/blob/master/core/src/main/java/org/apache/brooklyn/entity/group/AbstractMembershipTrackingPolicy.java#L242)

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java [hunk 2]
```diff
@@ -56,12 +60,17 @@ public void start(Collection<? extends Location> locations) {
...
             ((MongoDBRouterClusterImpl)super.entity).setAnyRouter();
         }
         @Override protected void onEntityRemoved(Entity member) {
+            // TODO shouldn't be invoked - remove
+            log.warn("Removal handler should be hidden by event handler", new Throwable("Trace for unexpected mongo node handler"));
```
Comment (tbouron) [inline software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: Not sure I understand what this is for: is this an error per se? Looks like a removal is handled correctly so why the warn message? Can you give a bit more details ?

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java [hunk 3]
```diff
@@ -56,12 +60,17 @@ public void start(Collection<? extends Location> locations) {
...
             ((MongoDBRouterClusterImpl)super.entity).setAnyRouter();
         }
         @Override protected void onEntityChange(Entity member) {
+            // TODO shouldn't be invoked - remove
+            log.warn("Change handler should be hidden by event handler", new Throwable("Trace for unexpected mongo node handler"));
```
Comment (tbouron) [inline software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: Same as above

File: software/webapp/src/main/java/org/apache/brooklyn/entity/dns/AbstractGeoDnsServiceImpl.java [hunk 4]
```diff
@@ -180,6 +180,7 @@ protected void endTracker() {
     public static class MemberTrackingPolicy extends AbstractMembershipTrackingPolicy {
         @Override
         protected void onEntityEvent(EventType type, Entity entity) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/webapp/src/main/java/org/apache/brooklyn/entity/dns/AbstractGeoDnsServiceImpl.java]: Not needed as this policy extends `AbstractMembershipTrackingPolicy ` and `onEntityEvent` [already call `defaultHighlightAction`](https://github.com/apache/brooklyn-server/blob/master/core/src/main/java/org/apache/brooklyn/entity/group/AbstractMembershipTrackingPolicy.java#L242)

File: software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java [hunk 5]
```diff
@@ -154,6 +154,7 @@ protected void removeServerPoolMemberTrackingPolicy() {
     public static class ServerPoolMemberTrackerPolicy extends AbstractMembershipTrackingPolicy {
         @Override
         protected void onEntityEvent(EventType type, Entity entity) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractControllerImpl.java]: Same as above

File: software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractNonProvisionedControllerImpl.java [hunk 6]
```diff
@@ -66,6 +67,7 @@ public AbstractNonProvisionedControllerImpl() {
 
     public static class MemberTrackingPolicy extends AbstractMembershipTrackingPolicy {
         @Override protected void onEntityEvent(EventType type, Entity member) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractNonProvisionedControllerImpl.java]: Same as above

File: software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/nginx/NginxControllerImpl.java [hunk 7]
```diff
@@ -178,6 +178,7 @@ protected void removeUrlMappingsMemberTrackerPolicy() {
     public static class UrlMappingsMemberTrackerPolicy extends AbstractMembershipTrackingPolicy {
         @Override
         protected void onEntityEvent(EventType type, Entity entity) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/nginx/NginxControllerImpl.java]: Same as above

File: software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractNonProvisionedControllerImpl.java [hunk 8]
```diff
@@ -216,6 +218,7 @@ protected void removeServerPoolMemberTrackingPolicy() {
     public static class ServerPoolMemberTrackerPolicy extends AbstractMembershipTrackingPolicy {
         @Override
         protected void onEntityEvent(EventType type, Entity entity) {
+            defaultHighlightAction(type, entity);
```
Comment (tbouron) [inline software/webapp/src/main/java/org/apache/brooklyn/entity/proxy/AbstractNonProvisionedControllerImpl.java]: Same as above

File: qa/src/test/java/org/apache/brooklyn/qa/load/AbstractLoadTest.java [hunk 9]
```diff
@@ -161,8 +161,8 @@ protected ManagementContext setUpPlatform() {
         // Create management node
         persistenceDir = Files.createTempDir();
         launcher = BrooklynLauncher.newInstance()
-                .persistMode(PersistMode.CLEAN)
-                .highAvailabilityMode(HighAvailabilityMode.MASTER)
+                .persistMode(doPersistence() ? PersistMode.CLEAN : PersistMode.DISABLED)
+                .highAvailabilityMode(doPersistence() ? HighAvailabilityMode.MASTER : HighAvailabilityMode.DISABLED)
```
Comment (tbouron) [inline qa/src/test/java/org/apache/brooklyn/qa/load/AbstractLoadTest.java]: I think for a flag, a field would be cleaner than a method to override

Comment (ahgittin) [reply qa/src/test/java/org/apache/brooklyn/qa/load/AbstractLoadTest.java]: due to how tests are structured and annotated methods run by the framework, methods for this sort of thing are easier to work with i find.  happy if you want to spike an alternative with fields but i doubt it's better.

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java (hunk 1 above)
Comment (ahgittin) [reply software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: this method overrides that method and doesn't call `super` so it has to redo it here, doesn't it?

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java (hunk 2 above)
Comment (ahgittin) [reply software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: more a note to developer (it caught me out) - due to the `onEntityEvent` handler being overridden this method is not used

**Comment by ahgittin** [comment]
thanks for the comments @tbouron - i've replied, and think the PR as it stands is correct.  wdyt?

**Comment by ahgittin** [comment]
Retest this please.

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java (hunk 1 above)
Comment (tbouron) [reply software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: Meh, you're right, I didn't see the missing `super` call for some reason.

File: software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java (hunk 2 above)
Comment (tbouron) [reply software/nosql/src/main/java/org/apache/brooklyn/entity/nosql/mongodb/sharding/MongoDBRouterClusterImpl.java]: That make sense now, as `onEntityEvent` does not call `super`

File: qa/src/test/java/org/apache/brooklyn/qa/load/AbstractLoadTest.java (hunk 9 above)
Comment (tbouron) [reply qa/src/test/java/org/apache/brooklyn/qa/load/AbstractLoadTest.java]: Good point, didn't think of the complexity due to the framework and annotation. Your solution works so let's go with that, we can iterate on it later on

**Comment by tbouron** [comment]
@ahgittin Yup, looks good but I would like Jenkins to be happy before merging it.

Looking at the logs, looks like the failure is unrelated though

**Comment by ahgittin** [comment]
hmm, failure is related:

```
[ERROR] /home/jenkins/jenkins-slave/workspace/brooklyn-library-pull-requests/software/network/src/test/java/org/apache/brooklyn/entity/network/bind/PrefixAndIdEnricher.java:[56,9] cannot find symbol
  symbol:   method highlightTriggers(org.apache.brooklyn.api.sensor.AttributeSensor<capture#1 of ?>,<nulltype>)
  location: class org.apache.brooklyn.entity.network.bind.PrefixAndIdEnricher
```

it's trying to reference `org.apache.brooklyn:brooklyn-parent:0.12.0-SNAPSHOT`.  have merged in master branch which changes deps to 0.13.0, retest will now hopefully be happy.

**Comment by ahgittin** [comment]
tests good after merging master, so merging this

**Pull request merged**